components/gitpod-db/src/typeorm/entity/db-workspace-instance.ts

```diff
@@ -96,4 +96,7 @@ export class DBWorkspaceInstance implements WorkspaceInstance {
 
     @Column("simple-json", { nullable: true })
     imageBuildInfo?: ImageBuildInfo;
+
+    @Column({ nullable: true })
+    workspaceClass?: string;
```

Comment:
If you add a field here, you'll also want to add a migration file:
- See [How to add a migration file](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/src/typeorm/migration/README.md) and [an example migration file](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/src/typeorm/migration/1653983151212-TS2MoreResources.ts)

At this point, you can choose whether you want this field to be possibly `NULL`, or if you want to set every row to a default value during the migration (e.g. to `'default'` or empty string `''`).

---

Another thing to consider is: How to we specify all the workspace classes that exist?
- We can either specify them as "any string" (current implementation)
- Or, we can specify them as an enum of only specific string values (for example, like [WorkspaceType](https://github.com/gitpod-io/gitpod/blob/aafb99b6ba90658679ee9c5febe762881cb71c59/components/gitpod-protocol/src/protocol.ts#L566))
  - the problem being that someone installing Gitpod (e.g. Self-Hosted) might define any number of classes with weird names (e.g. we currently define only [gitpodio-internal-xl](https://github.com/gitpod-io/ops/blob/b980015f8efeaf900c24970f2e1c095624793f80/deploy/workspace/installer-values.yaml#L40) but we could add more and rename them in the future 💭)
  - however, agreeing on a common set of values will be necessary e.g. to correctly aggregate usage hours in our Usage component 💡

Personally, I'd love to go with an enum similar to this one:

```ts
type WorkspaceClass = "" | "S" | "M" | "L" | "XL";
```

(with `""` being the default value that we set every row to in our migration) but feel free to make a call here. We can always adjust later.

Reply:
As discussed offline, the main issue is that we would need to keep this enum code in sync with whatever is available in the accompanying workspace-cluster, that in self-hosted might have arbitrarily named workspace classes.

We agreed for now to keep this as is, schedule a meeting to discuss how will WebApp's dashboard learn about the available classes https://github.com/gitpod-io/gitpod/issues/8261, and reevaluate.

Reply:
For more context, and also relevant context for usage/billing, we'll need a service to list available workspace types - this needs to differ per installation (or also possibly by a cohort/project which has access to a particular type). It makes perfect sense to store and define as string because it will vary beyond a fixed set of values.